build/ci/send-to-helix.yml

```diff
@@ -0,0 +1,58 @@
+  HelixConfiguration: ''                 # optional -- additional property attached to a job
+  IncludeDotNetCli: true                 # optional -- true will download a version of the .NET CLI onto the Helix machine as a correlation payload; requires DotNetCliPackageType and DotNetCliVersion
+  EnableXUnitReporter: true              # optional -- true enables XUnit result reporting to Mission Control
+  WaitForWorkItemCompletion: true        # optional -- true will make the task wait until work items have been completed and fail the build if work items fail. False is "fire and forget."
+  HelixBaseUri: 'https://helix.dot.net/' # optional -- sets the Helix API base URI (allows targeting int)
+  Creator: 'ML.NET'                      # optional -- if the build is external, use this to specify who is sending the job
+  DisplayNamePrefix: 'Run Helix Tests'   # optional -- rename the beginning of the displayName of the steps in AzDO
+  continueOnError: false                 # optional -- determines whether to continue the build if the step errors; defaults to false
+  condition: succeeded()                 # optional -- condition for step to execute; defaults to succeeded()
+
+steps:
+  - powershell: '$env:Path = "$env:BUILD_SOURCESDIRECTORY\.dotnet;$env:Path"; powershell "$env:BUILD_SOURCESDIRECTORY\eng\common\msbuild.ps1 $env:BUILD_SOURCESDIRECTORY\eng\helix.proj /t:Test /bl:$env:BUILD_SOURCESDIRECTORY\artifacts\log\${{ parameters.Configuration }}\SendToHelix.binlog /p:Configuration=${{ parameters.Configuration }} /p:TargetArchitecture=${{ parameters.Architecture }} -warnAsError 0"'
+    displayName: ${{ parameters.DisplayNamePrefix }} (Windows)
+    env:
+      BuildConfig: ${{ parameters.Configuration }}
+      BuildArchitecture: ${{ parameters.Architecture }}
+      HelixSource: ${{ parameters.HelixSource }}
+      HelixType: ${{ parameters.HelixType }}
+      HelixBuild: ${{ parameters.HelixBuild }}
+      HelixConfiguration:  ${{ parameters.HelixConfiguration }}
+      HelixTargetQueues: ${{ parameters.HelixTargetQueues }}
+      HelixAccessToken: ${{ parameters.HelixAccessToken }}
+      IncludeDotNetCli: ${{ parameters.IncludeDotNetCli }}
+      EnableXUnitReporter: ${{ parameters.EnableXUnitReporter }}
+      WaitForWorkItemCompletion: ${{ parameters.WaitForWorkItemCompletion }}
+      HelixBaseUri: ${{ parameters.HelixBaseUri }}
+      SYSTEM_ACCESSTOKEN: $(System.AccessToken)
+      Creator: ${{ parameters.Creator }}
+    condition: and(${{ parameters.condition }}, eq(variables['Agent.Os'], 'Windows_NT'))
+    continueOnError: ${{ parameters.continueOnError }}
+  - script: export PATH=$BUILD_SOURCESDIRECTORY/.dotnet:$PATH && $BUILD_SOURCESDIRECTORY/eng/common/msbuild.sh $BUILD_SOURCESDIRECTORY/eng/helix.proj /t:Test /bl:$BUILD_SOURCESDIRECTORY/artifacts/log/${{ parameters.Configuration }}/SendToHelix.binlog /p:Configuration=${{ parameters.Configuration }} /p:TargetArchitecture=${{ parameters.Architecture }} --warnAsError false
```

Comment:
Usually when having an entry for Windows and an entry for Unix it is hard to maintain both up to date when a new parameter is added. I would suggest instead taking as a parameter the build script and then the environment variables set then under the `env:` task parameter.

For example see: https://github.com/dotnet/runtime/blob/98be10a336f216127470214fedf928ba68635c60/eng/pipelines/libraries/helix.yml#L18

In that example the command is a variable we set at the job level rather than a parameter:
https://github.com/dotnet/runtime/blob/b482c04080bb86f151e8fe3d6f2fe61649c1b5fa/eng/pipelines/common/xplat-setup.yml#L62-L63
https://github.com/dotnet/runtime/blob/b482c04080bb86f151e8fe3d6f2fe61649c1b5fa/eng/pipelines/common/xplat-setup.yml#L78-L79

Reply:
I did it this way because its the way the `send-to-helix.yml` file from Arcade is done and I just followed their example. For the actual build command we do it the way you suggest though. If you feel strongly that I should change it I will, but I think leaving it as close to their example as possible will be easier for others who aren't as familiar with the process to understand whats going on.

Reply:
The reason why the arcade `send-to-helix.yml` is like that, is because it is a more generic file for repos to share and use, so it would be harder user experience to use a predefined variable because for templates you usually just read parameters. I did propose that template to take a parameter for the msbuild script command, however they wanted to keep it simple and repos to not have to think about checking for windows vs unix. 

However, since this is your custom send to helix, I would definitely encourage a cleaner/smaller approach rather than duplicating, it helps with readability and maintainability in the future when you add more properties that need to be set for the helix proj.

Reply:
That makes sense. I'll go ahead and switch it.